lib/IRGen/GenHeap.cpp

```diff
@@ -1900,7 +1892,7 @@ IsaEncoding irgen::getIsaEncodingForType(IRGenModule &IGM,
 
   if (auto theClass = type->getClassOrBoundGenericClass()) {
     // We can access the isas of pure Swift classes directly.
-    if (getRootClass(theClass)->hasKnownSwiftImplementation())
+    if (!theClass->checkAncestry().contains(AncestryFlags::HasImported))
```

Comment:
Can you throw a convenience method on `ClassDecl` that does this?  It seems to be repeated.

Reply:
Yeah, that's a good idea.